meilisearch-core/src/store/facets.rs

```diff
@@ -0,0 +1,57 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use heed::{types::CowSlice, RwTxn, RoTxn, Result as ZResult};
+use sdset::{SetBuf, Set, SetOperation};
+
+use meilisearch_types::DocumentId;
+
+use crate::database::MainT;
+use crate::error::Error;
+use crate::facets::FacetKey;
+
+/// contains facet info
+#[derive(Clone, Copy)]
+pub struct Facets {
+    pub(crate) facets: heed::Database<FacetKey, CowSlice<DocumentId>>,
+}
+
+impl Facets {
+    // we use sdset::SetBuf to ensure the docids are sorted.
+    pub fn put_facets(&self, writer: &mut RwTxn<MainT>, facet_key: FacetKey, doc_ids: &Set<DocumentId>) -> ZResult<()> {
+        self.facets.put(writer, &facet_key, doc_ids)
+    }
+
+    pub fn document_ids<'txn>( &self, reader: &'txn RoTxn<MainT>, facet_key: &FacetKey,) -> Result<Option<Cow<'txn, [DocumentId]>>, Error> {
```

Comment:
```suggestion
    pub fn facet_ids<'txn>( &self, reader: &'txn RoTxn<MainT>, facet_key: &FacetKey) -> Result<Option<Cow<'txn, [DocumentId]>>, Error> {
```

Reply:
BTW I don't unsderstand why this function doesn't just return a `ZResult<Cow<'txn, Set<DocumentId>>>`?

Reply:
Yeah, we should store a `Set<DocumentId>` because the purpose of these ids is to perform set operations on them. Also I think you are right and we could call this function `facet_documents_ids`, it fit better.

Reply:
How can I get heed to return me a Cow<Set<DocumentId>>??